### ISSUE TITLE
Buffer out of range in node 10

### ISSUE THREAD

**Issue by soyuka** [theturtle32/WebSocket-Node#334]
Hi guys,

I have the following range error with node 10:
```
RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received -903354019
    at checkInt (internal/buffer.js:35:11)
    at writeU_Int32BE (internal/buffer.js:623:3)
    at Buffer.writeUInt32BE (internal/buffer.js:636:10)
    at WebSocketFrame.toBuffer (/home/travis/build/soyuka/signalhubws/node_modules/websocket/lib/WebSocketFrame.js:257:24)
    at WebSocketConnection.sendFrame (/home/travis/build/soyuka/signalhubws/node_modules/websocket/lib/WebSocketConnection.js:857:43)
    at WebSocketConnection.fragmentAndSend (/home/travis/build/soyuka/signalhubws/node_modules/websocket/lib/WebSocketConnection.js:793:14)
    at WebSocketConnection.sendUTF (/home/travis/build/soyuka/signalhubws/node_modules/websocket/lib/WebSocketConnection.js:733:10)
    at W3CWebSocket.send (/home/travis/build/soyuka/signalhubws/node_modules/websocket/lib/W3CWebSocket.js:116:26)
```

The origin call is something like: 

```
socket.send(JSON.stringify(data))
```

Not sure what's wrong because this worked just fine with node < 10.

Related issue: https://github.com/soyuka/signalhubws/pull/6

**Comment by sundagy** [comment]
Got same error.

Here th part of code with problem:
lib/WebSocketFrame.js:257
```
        maskKey = nullMask ? 0 : ((Math.random() * 0xFFFFFFFF) >>> 0);
        this.maskBytes.writeUInt32BE(maskKey, 0);
```

**Comment by aeisenberg** [comment]
What version are you using? @sundagy and @soyuka? We have been seeing the same problem in v1.0.25. The code looks slightly different there:

        maskKey = nullMask ? 0 : (Math.random()*0xFFFFFFFF) | 0;
        this.maskBytes.writeUInt32BE(maskKey, 0, true);

Looks like the `|` was changed to `>>>`. Indeed, `|` looks incorrect since that doesn't create an unsigned 32bit integer. It's likely that node 10 is stricter than node 8, since there were a lot of Buffer changes in that release.

So, my guess is that this error only occurs on earlier versions of websocket 1.0.25 and before. Is that your experience?

We've downgraded to node v8 for now, and when we have time again, we will upgrade to websocket v1.0.28 and see if we still get the error.
